### PR TITLE
fix: shouldResetPlayer options spread

### PR DESCRIPTION
**Pull request by staff0rd**
Reset is currently broken, mostly due to the position of the `playerVars` spread. I also added a test to prove this.

Prior to this fix, any change to `start` was still resulting in a player reset.

### CODE REVIEW

**Review by ruisaraiva19 — APPROVED** [review]

**Comment by ruisaraiva19** [comment]
@staff0rd This looks good. Let's wait for #278 to be merged first.

**Review by tjallingt — APPROVED** [review]
Looks great, thanks!

**Comment by ruisaraiva19** [comment]
@staff0rd can you sync your fork with the latest changes from this repo before I can merge this?

**Comment by iyzana** [comment]
@staff0rd Do you still know why you added `prevProps.videoId !== props.videoId` to `shouldResetPlayer`?
From what I've observed, changing just the videoId works perfectly fine without resetting the player.
With this check the player unnecessarily leaves fullscreen when the video changes.

**Pull request merged**
